src/client/src/ui/spotTile/model/spotTileModel.js

```diff
@@ -113,22 +114,29 @@ export default class SpotTileModel extends ModelBase {
       let request = this._createTradeRequest(e.direction);
       this._log.info(`Will execute ${request.toString()}`);
       this.isTradeExecutionInFlight = true;
+      window.clearTimeout(this.notificationTimer);
       this._executionDisposable.setDisposable(
-        this._executionService.executeTrade(request).subscribeWithRouter(
-          this.router,
-          this.modelId,
-          (response:ExecuteTradeResponse) => {
-            this.isTradeExecutionInFlight = false;
-            this.notification = response.hasError
-              ? TradeExecutionNotification.createForError(response.error)
-              : TradeExecutionNotification.createForSuccess(response.trade);
-          },
-          err => {
-            this.isTradeExecutionInFlight = false;
-            this._log.error(`Error executing ${request.toString()}. ${err}`, err);
-            this.notification = TradeExecutionNotification.createForError(`Unknown stream error`);
+        this._executionService.executeTrade(request)
+          .do(response => {
+            if (!response.hasError && response.trade.status === TradeStatus.Done) {
+              this.notificationTimer = setTimeout(() => this.router.publishEvent(this.modelId, 'tradeNotificationDismissed', {}), DISMISS_NOTIFICATION_AFTER_X_IN_MS);
```

Comment:
Can we use the rx schedule server rather than setTimeout?

It'll wrap setTimeout in a nicer api that we can easily dispose. for example something like this:

```
this.addDisposable(_scheduleService.async(1000, () => { ... })
```

I can't remember the actual method names, but that's close